sql/database_cache.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import "time"
+
+// DBConnCache caches the DB object
+type DBConnCache struct {
```

Comment:
FYI, the Hive driver and ODPS driver doesn't maintain a long connection on each `DB` object. So adding cache may not help.

Reply:
Not only the performance,  to cache the `DB` object can save the SQL context, such as `USE DB1` if we can load balance the requests to the same SQLFlow server instance according to the `Token`. And as the comment: https://github.com/sql-machine-learning/sqlflow/pull/530#discussion_r296898832 ,  I will remove the `db_conn_cache` from this PR, and do some benchmarks.